### ISSUE TITLE
Face recognized and classes allocated, still no module reveal.

### ISSUE THREAD

**Issue by SdeGeata** [normyx/MMM-Facial-Recognition-OCV3#31]
Greetings;

Well after two weeks of trying to install Facial Recognition on my MM, it all works. Now one last final bug to squash and I should be able to reap the rewards.

So Facial Recognition is loading fine and my modules have the appropriate classes assigned. Terminal output reads that I am indeed recognized;

`[MMM-Facial-Recognition-OCV3] User Stiofan with confidence 70.7123926022 logged in.
`

However, none of the modules with classes assigned to my profile are revealed. Any idea why???

**Comment by SdeGeata** [comment]
So really, nobody???

**Comment by mraahul** [comment]
Hey @SdeGeata, I am working on this as a part of my capstone project and facing the same issue as you were. Were you able to resolve it? Did you get to know what was the reason?

Any help is greatly appreciated. Thanks. Cheers :)

**Comment by SdeGeata** [comment]
Hello Mraahul!

No, still no success. I have been trying in vain for quite some time and have had very little luck.

I did have some success very recently with Nischi’s MMM-Face-Reco-DNN. He has been quite helpful in attempting to stomp out issues and getting it working. His module actually did work for me, briefly. Then i got a notification in MM to update a few things and the updates broke it. I am presently attempting to troubleshoot what went wrong with that, but i recommend giving that module a try.

**Comment by mraahul** [comment]
Thanks for your instant reply @SdeGeata. So you were able to see the other modules while the Facial recognition was done using Nischi’s MMM-Face-Reco-DNN? If yes, do you think it will work with Rpi 3b+ ?

I want to do it this way -> when the face is detected, screen will show all the default modules. Like our cellphones, FaceID to unlock the phone.

**Comment by SdeGeata** [comment]
Yeah, i’m using a Rpi 3b+. Like i said, the module works fine... i just gotta figure out the conflict. It is class based so whatever module you assign that class to, the assigned face will unlock.

I have modules assigned to myself and my girlfriend, as well as default modules that anyone can see. Nischi’s module is the only face recognition module that i had any luck with, but i introduced some conflict when i updated another module and Rpi stretch; now it doesn't work. I’m troubleshooting now and ultimately may start from scratch.

**Comment by mraahul** [comment]
@SdeGeata  I was looking around the net to see how the classes are configured and I came across this guy who had a different issue but was able to define the modules as per the classes. I used his technique which solved my problem. I hope this resolves yours too.

https://forum.magicmirror.builders/topic/5522/mmm-facial-recognition-and-mmm-assistant-don-t-seem-to-be-working-together/3

**Comment by SdeGeata** [comment]
Glad your problem is fixed. Mine is a whole other issue, gonna start over from scratch.

**Comment by FascinatedTV** [comment]
Hi @SdeGeata, faced this issue myself. If you still interested in a possible solution try start magic mirror in dev
(npm run start:dev). On face recognition i got a javascript error in a file provided by the magic mirror repo.
Resolving the error [MagicMirror/js/main.js](https://github.com/MichMich/MagicMirror/blob/master/js/main.js) Line 415 (modulesByClass) worked for me.

**Comment by SdeGeata** [comment]
Hey there!   Thanks for your reply. I no longer have this issue, i am using Nischi’s MMM-Face-Reco-DNN and that appears to have solved that issue.On Jan 11, 2023, at 1:55 PM, FascinatedTV ***@***.***> wrote:﻿
Hi @SdeGeata, faced this issue myself. If you still interested in a possible solution try start magic mirror in dev
(npm run start:dev). On face recognition i got a javascript error in a file provided by the magic mirror repo.
Resolving the error MagicMirror/js/main.js Line 415 (modulesByClass) worked for me.
